environment/src/main/scala/ru/primetalk/funtik/environment/genereator/Tree.scala

```diff
@@ -0,0 +1,7 @@
+package ru.primetalk.funtik.environment.genereator
+
+
+sealed trait Tree[T]
```

Comment:
What do you think about "eliminator":
```scala
object Tree {
  def eliminate[T, A](t: Tree[T])(node: (A, A) => A))(leaf: T => A): A = t match {
    case Node(l, r) => node( eliminate(l)(node)(leaf), eliminate(r)(node)(leaf) )
    case Leaf(v) => leaf(v)
  }
}
```
?
We could use this in any algorithm instead of pattern matching.